docs/datamodel/index.rst

```diff
@@ -1,50 +1,116 @@
+
+Databases
+=========
+
+An EdgeDB instance can have multiple databases in it. Every instance
+is created with an empty database called "edgedb". If you wish to add
+more, you can use the :eql:stmt:`CREATE DATABASE` EdgeQL command.
+Conversely, the :eql:stmt:`DROP DATABASE` command removes a database.
+
+The following command will get a list of all databases present in the
+instance:
+
+.. code-block:: edgeql
+
+    SELECT sys::Database.name;
+
+.. _ref_datamodel_modules:
+
+Modules
+=======
+
+Every database has a schema that fully describes it. Schema consists
+of logical units called  **modules**, which act as namespaces.
+Modules contain user-defined types, functions, etc. A module has a
+name that is unique inside a database. The same schema object name can
+be used in different modules without conflict.  For example, both
+``module1`` and ``module2`` can contain a ``User`` object type.
+
+Schema objects can be referred to by a fully-qualified name using the
+``<module>::<name>`` notation.
+
+Every EdgeDB schema contains the following standard modules:
+
+* ``std``: standard types, functions and other declarations
+* ``schema``: types describing the :ref:`introspection <ref_eql_introspection>`
+  schema
+* ``sys``: system-wide entities, such as user roles and
+  :ref:`databases <ref_datamodel_databases>`
+* ``cfg``: configuration and settings
+* ``stdgraphql``: GraphQL-related types
+* ``math``: algebraic and statistical functions
+* ``default``: the default module for user-defined types, functions, etc.
+
+EdgeDB provides :ref:`migration tools <ref_cli_edgedb_migration>` that
+use a a high-level declarative :ref:`schema definition
+language<ref_eql_sdl>` to manage the schema state. The EdgeDB SDL is
+designed to be a concise and readable representation of the schema.
+Most of the examples and synopses in this documentation use the SDL
+notation.
+
+Here's a quick overview of the kind of things that can be defined in
+modules:
+
+Types in EdgeDB include your own **Object Types** (e.g. *User*) and
+**Abstract Types** for other types to extend (e.g. ``HasEmailAddress``
+for ``User`` and others can inherit), plus **Scalar Types** with
+single values (``str``, ``int64``, etc.) and **Collection Types** like
+**arrays** and **tuples** for multiple values.
+
+Putting your **Object Types** together are **properties** and
+**links**. You can build on them with items like **annotations**
+(readable notes for others), **constraints** to set limits (e.g.
+maximum length, minimum value, or even create your own), **indexes**
+for faster querying, and **computed** properties or links to define
+useful expressions (e.g. ``property email := .user_name ++ '@' ++
+.provider_name``).
+
+**Expression Aliases** let you use existing types under new names to
+build on them without touching the original -- both in your schema or
+on the fly inside a query.
+
+You can also create your own **functions**, strongly typed along with
+everything else in EdgeDB.
```

Comment:
Love this new section, great job. Now we just need a cool diagram, but we'll add it later.